tiledb/api/c_api/array/test/unit_capi_array.cc

```diff
@@ -30,78 +30,1345 @@
+#include "tiledb/api/c_api_test_support/testsupport_capi_context.h"
 
 using namespace tiledb::api::test_support;
 
 const char* TEST_URI = "unit_capi_array";
 
+TEST_CASE(
+    "C API: tiledb_array_schema_load argument validation", "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_array_schema_t* schema{};
+  /*
+   * No "success" section here; too much overhead to set up.
+   */
+  SECTION("null context") {
+    rc = tiledb_array_schema_load(nullptr, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null uri") {
+    rc = tiledb_array_schema_load(ctx.context, nullptr, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_schema_load(ctx.context, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_schema_free(&schema));
+  CHECK(schema == nullptr);
+}
+
+TEST_CASE(
+    "C API: tiledb_array_schema_load_with_config argument validation",
+    "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_config_t* config{};
+  tiledb_array_schema_t* schema{};
+  /*
+   * No "success" section here; too much overhead to set up.
+   */
+  SECTION("null context") {
+    rc = tiledb_array_schema_load_with_config(
+        nullptr, config, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null config") {
+    // Note: a null config is actually valid and will use the context's config.
+    // This test case merely fails without the proper overhead setup.
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, nullptr, TEST_URI, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null uri") {
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, config, nullptr, &schema);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_schema_load_with_config(
+        ctx.context, config, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_schema_free(&schema));
+  CHECK(schema == nullptr);
+}
+
+TEST_CASE("C API: tiledb_array_alloc argument validation", "[capi][array]") {
+  capi_return_t rc;
+  ordinary_context ctx{};
+  tiledb_array_handle_t* array{};
+  SECTION("success") {
+    rc = tiledb_array_alloc(ctx.context, TEST_URI, &array);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+  }
+  SECTION("null context") {
+    rc = tiledb_array_alloc(nullptr, TEST_URI, &array);
+    REQUIRE(tiledb_status(rc) == TILEDB_INVALID_CONTEXT);
+  }
+  SECTION("null uri") {
+    rc = tiledb_array_alloc(ctx.context, nullptr, &array);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null schema") {
+    rc = tiledb_array_alloc(ctx.context, TEST_URI, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  REQUIRE_NOTHROW(tiledb_array_free(&array));
```

Comment:
This needs to be in the "success" test and not elsewhere. If it's not open, it shouldn't be freed. It's only not failing here because all the `*_free` functions return `void` and not `capi_return_t`.